app/src/components/evm-debugger/LogsAndReturn.tsx

```
@@ -0,0 +1,168 @@
+          >
+            <div class="divide-y divide-gray-100 dark:divide-gray-800">
+              <For each={props.state.logs}>
+                {(log, index) => (
+                  <div class="px-4 py-2.5 hover:bg-gray-50 dark:hover:bg-[#2D2D2D] transition-colors group">
+                    <div class="flex justify-between items-center">
+                      <span class="text-xs font-medium text-gray-500 dark:text-gray-400">Log {index()}:</span>
+                      <button
+                        onClick={() => handleCopyLog(log, index())}
+                        class="opacity-0 group-hover:opacity-100 transition-opacity text-gray-400 hover:text-gray-600 dark:hover:text-gray-300 p-1 rounded-md hover:bg-gray-100 dark:hover:bg-gray-700"
+                        aria-label="Copy to clipboard"
+                      >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)